fabric_generator/fabric_gen.py

```diff
@@ -15,6380 +15,2278 @@
+                input, output = i.expandPortInfo("AutoSwitchMatrix")
+                sourceName += input
+                destName += output
+        # bel wire
+        for b in tile.bels:
+            for p in b.inputs:
+                sourceName.append(f"{p}")
+            for p in b.outputs:
+                destName.append(f"{p}")
+
+        # jump wire
+        for i in tile.portsInfo:
+            if i.wireDirection == Direction.JUMP:
+                input, output = i.expandPortInfo("AutoSwitchMatrix")
+                sourceName += input
+                destName += output
+        sourceName = list(dict.fromkeys(sourceName))
+        destName = list(dict.fromkeys(destName))
+        # get indexed version of the port of the tile
+        for p in tile.portsInfo:
+            if p.wireDirection != Direction.JUMP:
+                input, output = p.expandPortInfo(
+                    mode="AutoSwitchMatrixIndexed")
+                portInputIndexed += input
+                portOutputIndexed += output
+                input, output = p.expandPortInfo(mode="AutoTopIndexed")
+                portTopInput += input
+                portTopOutputs += output
+            else:
+                input, output = p.expandPortInfo(
+                    mode="AutoSwitchMatrixIndexed")
+                input = [i for i in input if "GND" not in i and "VCC" not in i]
+                jumpWire += input
+
+        belOutputs = []
+        belInputs = []
+        # removing duplicates
+        portOutputIndexed = list(dict.fromkeys(portOutputIndexed))
+        portTopInput = list(dict.fromkeys(portTopInput))
+        jumpWire = list(dict.fromkeys(jumpWire))
+        for b in tile.bels:
+            belOutputs += b.outputs
+            belInputs += b.inputs
+
+        portList = [i for i in (destName + sourceName)
+                    if "GND" not in i and "VCC" not in i]
+        signal = portOutputIndexed + belOutputs + \
+            jumpWire + portTopInput + belInputs + jumpWire
+
+        configBit = ""
+        if self.fabric.configBitMode == ConfigBitMode.FLIPFLOP_CHAIN:
+            portList.append("MODE")
+            portList.append("CONFin")
+            portList.append("CONFout")
+            portList.append("CLK")
+            signal.append("Mode")
+            signal.append(f"conf_data[{belCounter}]")
+            signal.append(f"conf_data[{belCounter+1}]")
+            signal.append("CLK")
+        if self.fabric.configBitMode == ConfigBitMode.FRAME_BASED:
+            if tile.globalConfigBits > 0:
+                portList.append("ConfigBits")
+                portList.append("ConfigBits_N")
+                signal.append(
+                    f"ConfigBits[{tile.globalConfigBits}-1:{belConfigBitsCounter}]")
+                signal.append(
+                    f"ConfigBits_N[{tile.globalConfigBits}-1:{belConfigBitsCounter}]")
+
+        self.writer.addInstantiation(compName=f"{tile.name}_SwitchMatrix",
```

Comment:
case: this is being generated as `<tile>_switch_matrix` but then instantiated as `<tile>_SwitchMatrix`